apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/bigint.ex

```diff
@@ -0,0 +1,46 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.DataUpdaterPlant.DataUpdater.BigInt do
+  @moduledoc """
+  Ecto type for Cassandra/Scylla bigints, aka 64-bit signed longs.
+  """
+  use Ecto.Type
+
+  @type t() :: integer()
+
+  @min_bigint -2 ** 63
+  @max_bigint 2 ** 63 - 1
+
+  @doc false
+  @impl Ecto.Type
+  def type, do: :bigint
+
+  @impl Ecto.Type
+  @spec load(t() | any()) :: {:ok, t()} | :error
+  def load(n) when is_integer(n) and n >= @min_bigint and n <= @max_bigint, do: {:ok, n}
+  def load(_), do: :error
```

Comment:
We should also handle the `nil` case, without returning an error.
Same thing for the SmallInt type.

Reply:
Actually, Ecto's documentation reports:

> Note: nil values are always bypassed and cannot be handled by custom types.

So it's fine like this.